cmd/manager/main.go

```diff
@@ -147,64 +144,64 @@ func main() {
 	if clusterOperatorName != "" {
 		statusReporter, err = status.NewReporter(cfg, mgr, namespace, clusterOperatorName, os.Getenv("RELEASE_VERSION"), stopCh)
 		if err != nil {
-			exit(err)
+			logrus.Fatal(err)
 		}
 	}
 
 	// Populate the global default OperatorSources definition and config
 	err = defaults.PopulateGlobals()
 	if err != nil {
-		exit(err)
+		logrus.Fatal(err)
 	}
 
 	// Setup all Controllers
 	if err := controller.AddToManager(mgr, options.ControllerOptions{}); err != nil {
-		exit(err)
+		logrus.Fatal(err)
 	}
 
-	// Serve a health check.
-	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(http.StatusOK)
-	})
-	go http.ListenAndServe(":8080", nil)
+	logrus.Info("Setting up health checks")
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
+		logrus.Fatal(err, "unable to set up health check")
+	}
+	if err := mgr.AddReadyzCheck("healthz", healthz.Ping); err != nil {
+		logrus.Fatal(err, "unable to set up ready check")
+	}
 
 	// Wait until this instance becomes the leader.
-	log.Info("Waiting to become leader.")
-	err = leader.Become(context.TODO(), "marketplace-operator-lock")
+	logrus.Info("Waiting to become leader.")
+	err = leader.Become(context.TODO(), leaderElectionConfigMapName)
 	if err != nil {
-		log.Error(err, "Failed to retry for leader lock")
-		os.Exit(1)
+		logrus.Fatal(err, "Failed to retry for leader lock")
 	}
-	log.Info("Elected leader.")
+	logrus.Info("Elected leader.")
 
-	log.Print("Starting the Cmd.")
+	logrus.Info("Starting the Cmd.")
 
 	// migrate away from Marketplace API
 	clientGo, err := client.New(cfg, client.Options{Scheme: mgr.GetScheme()})
 	if err != nil && !k8sErrors.IsNotFound(err) {
-		log.Error(err, "Failed to instantiate client for migrator")
-		os.Exit(1)
+		logrus.Fatal(err, "Failed to instantiate the client for migrator")
 	}
 	migrator := migrator.New(clientGo)
 	err = migrator.Migrate()
 	if err != nil {
-		log.Error(err, "[migration] Error in migrating Marketplace away from OperatorSource API")
```

Comment:
It looks like some of these log calls were accidentally changed to fatal (https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/operator-framework_operator-marketplace/391/pull-ci-operator-framework-operator-marketplace-master-e2e-aws/1389397643536371712/artifacts/e2e-aws/gather-extra/artifacts/pods/openshift-marketplace_marketplace-operator-6c6799c7cc-9r89t_marketplace-operator.log).

Reply:
Fyi we can even get rid of the migrator now, can be a separate PR too but I'm not opposed to getting it done in this PR (since we need a separate bz for it if we want this in 4.8)

Reply:
@benluddy Thanks, that was a poor replacement job. I updated a couple of instances from Fatal back to Error.

@anik120 I'm down with removing it in this PR or adding a TODO comment saying this can be removed. I feel like removing the migrator should be verified separately though, so we may want to pursue the latter option.

Reply:
Actually, is the migrator removable? Are we still backwards compatible with earlier versions of OCP without it?

Reply:
This migrator was for migrating any remaining OperatorSources to CatalogSources in 4.5(or 4.6, one of those). Now that we don't have the OperatorSource, all it does it tries to get the OperatorSources in the cluster, but encounters the error that there's no CRD called OperatorSource, and logs `[migration] Error in migrating Marketplace away from OperatorSource API` and exits out. 

> I feel like removing the migrator should be verified separately though, so we may want to pursue the latter option.

Totally understandable if we don't feel comfortable doing this now. FWIW though, given what I've said above, it's a pretty innocuous change, at least from the dev team's perspective.

Reply:
Let's keep that separate, this is already a substantial PR.